### PR TITLE
api: no dangling swap balance

### PR DESCRIPTION
**Pull request by dcposch**

### CODE REVIEW

**Comment by dcposch** [comment]
Shipped + tested > cc @andrewliu08 for post review

File: packages/daimo-api/src/db/db.ts [hunk 1]
```diff
@@ -179,183 +179,6 @@ export class DB {
...
-          from_amount NUMERIC NOT NULL,
-          to_chain_id NUMERIC NOT NULL,
-          to_addr BYTEA NOT NULL,
-          to_token BYTEA NOT NULL,
-          to_amount NUMERIC NOT NULL,
-          nonce NUMERIC NOT NULL,
-          new_recipient BYTEA,
-          final_recipient BYTEA,
-          PRIMARY KEY (chain_id, block_num, log_idx)
-        );
```
Comment (dcposch) [inline packages/daimo-api/src/db/db.ts:358]: moved to correct spot in #1238

File: packages/daimo-api/src/server/tokenRegistry.ts [hunk 2]
```diff
@@ -76,8 +76,8 @@ export class TokenRegistry {
           foreignTokens.set(addr, {
             token: addr,
             decimals: token.decimals,
-            name: token.name,
-            symbol: token.symbol,
+            name: token.name.trim().replaceAll(/  */g, " "),
```
Comment (dcposch) [inline packages/daimo-api/src/server/tokenRegistry.ts:79]: fixes some token names, including Degen > no trailing spaces

File: packages/daimo-api/src/contract/foreignCoinIndexer.ts [hunk 3]
```diff
@@ -152,12 +166,23 @@ export class ForeignCoinIndexer extends Indexer {
       );
 
       // Special case: we previously swapped ETH in bulk
-      // If there's an unmatched outbound, clear prior inbounds.
+      // If there's an unmatched outbound, replace all prior inbounds with a single one for the full balance.
```
Comment (dcposch) [inline packages/daimo-api/src/contract/foreignCoinIndexer.ts:169]: this ensures we don't have dangling, unclaimable token balances in people's V1 accounts

**Review by andrewliu08 — APPROVED** [review]

**Pull request merged**